runtime/lua/vim/shared.lua

```diff
@@ -976,4 +976,41 @@ do
   end
 end
 
+---@param tabpage tabpage
+---@param layout table
+function vim._set_layout(tabpage, layout)
+  local function set_layout(node)
+    local ty = node[1]
+    if ty == 'leaf' then
+      local buf = node[2]
+      if type(buf) == 'string' then
```

Comment:
Can use either a string to open a buffer, or a number to use an already-opened buffer. Setting buflisted manually is a bit hacky, but I don't know of a better way to open a buffer from Lua and get its bufnr. Could use input on that.